### PR TITLE
Use Node 22 (current LTS) for development

### PR DESCRIPTION
**Pull request by adamkudrna**
- Drop Node 18 for testing builds.
- Add Node 22.x to dev engines.

### CODE REVIEW

**Review by mbohal — APPROVED** [review]

**Comment by bedrich-schindler** [comment]
I would update package.json in such way, that I would remove `engines` section and introduce `devEngines`:

```
  "devEngines": {
    "node": "20.x || 22.x"
  },
  ```
  
  This is how React does it. RUI does not required any specific version to run, only dev environment does.
  
  What do you think @adamkudrna @mbohal ?

**Comment by mbohal** [comment]
I agree with using `devEngines`. However the code example provided by @bedrich-schindler is wrong.

See: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#devengines for usage. I wonder if we should fail or warn on error. I think I would prefer to fail…

**Comment by bedrich-schindler** [comment]
@mbohal good point. It is weird that library like React have it wrong, but you are right unless there is some undocumented approach.

File: package.json [hunk 1]
```diff
@@ -31,6 +31,17 @@
...
+    "runtime": {
+      "name": "node",
+      "version": "22.x",
+      "onFail": "error"
+    },
+    "packageManager": {
+      "name": "npm",
+      "onFail": "error"
+    }
+  },
```
Comment (adamkudrna) [inline package.json:44]: @bedrich-schindler @mbohal What do you think?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
       - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
-          node-version: 20
+          node-version: 22
```
Comment (adamkudrna) [inline .github/workflows/lint.yml:16]: Update commit message from "upcoming" to "current".

File: package.json (hunk 1 above)
Comment (adamkudrna) [reply package.json:44]: Approved by @bedrich-schindler.

**Pull request merged**